builtin/pure_ysh.py

```diff
@@ -102,6 +102,106 @@ def Run(self, cmd_val):
         return 0
 
 
+class ctx_Context(object):
+    """For ctx push (context) { ... }"""
+
+    def __init__(self, mem, context):
+        # type: (state.Mem, Dict[str, value_t]) -> None
+        self.mem = mem
+        self.mem.PushContextStack(context)
+
+    def __enter__(self):
+        # type: () -> None
+        pass
+
+    def __exit__(self, type, value, traceback):
+        # type: (Any, Any, Any) -> None
+        self.mem.PopContextStack()
+
+
+class Ctx(vm._Builtin):
+
+    def __init__(self, mem, cmd_ev):
+        # type: (state.Mem, CommandEvaluator) -> None
+        self.mem = mem
+        self.cmd_ev = cmd_ev  # To run blocks
+
+    def _GetContext(self):
+        # type: () -> Dict[str, value_t]
+        ctx = self.mem.GetContext()
+        if ctx is None:
+            raise error.FatalRuntime(
```

Comment:
I think maybe error.Expr is more conventional and will also use status 3

I think that is the "default" for errors in new YSH features and the ctx is an expression, so it seems to make sense